### PR TITLE
single binary, even for wrapper

### PR DESCRIPTION
**Pull request by teknoraver**
remove compatibility wrapper

### CODE REVIEW

**Comment by zyga** [comment]
This does look nice, thank you. Let me do some testing (we don't have any automated CI setup for this repository)

**Comment by teknoraver** [comment]
The argument shift seems to work:

```
root@ubuntu32:~/src/snap-confine# src/snap-confine arg1 arg2
appname arg1 not allowed
root@ubuntu32:~/src/snap-confine# compat/ubuntu-core-launcher arg_old arg1 arg2
appname arg1 not allowed
```

File: compat/Makefile.am [hunk 1]
```diff
@@ -1,7 +1,5 @@
 dist_bin_SCRIPTS = ubuntu-core-launcher
 CLEANFILES = $(dist_bin_SCRIPTS)
-EXTRA_DIST = ubuntu-core-launcher.in
 
-ubuntu-core-launcher: ubuntu-core-launcher.in Makefile
-	sed -e 's,[@]libexecdir[@],$(libexecdir),g' < $(srcdir)/ubuntu-core-launcher.in > ubuntu-core-launcher
-	chmod +x ubuntu-core-launcher
+ubuntu-core-launcher: Makefile
+	ln -s snap-confine ubuntu-core-launcher
```
Comment (zyga) [inline compat/Makefile.am:5]: Note that the location of snap-confine and ubuntu-core-launcher can be rather different. I will need to experiment with how this affects downstream distributions. Some packages (_ahem_ dpkg) don't deal very well with file/symlink transitions AFAIR

**Comment by mvo5** [comment]
Symlinks to files should be fine with dpkg. Symlinks to directories are problematic because dpkg tries to be smart about them

**Pull request merged**